py-polars/tests/unit/operations/test_cast.py

```diff
@@ -332,6 +338,26 @@ def _cast_lit_t(
         (timedelta(seconds=1), pl.Duration("ms"), pl.Int32, True, MS_PER_SECOND),
         (timedelta(seconds=1), pl.Duration("us"), pl.Int64, True, US_PER_SECOND),
         (timedelta(seconds=1), pl.Duration("ns"), pl.Int64, True, NS_PER_SECOND),
+        # duration to boolean
+        (timedelta(seconds=0), pl.Duration("ms"), pl.Boolean, True, False),
+        (timedelta(seconds=1), pl.Duration("ms"), pl.Boolean, True, True),
+        (timedelta(seconds=0), pl.Duration("us"), pl.Boolean, True, False),
+        (timedelta(seconds=1), pl.Duration("us"), pl.Boolean, True, True),
+        (timedelta(seconds=0), pl.Duration("ns"), pl.Boolean, True, False),
+        (timedelta(seconds=1), pl.Duration("ns"), pl.Boolean, True, True),
+        # datetime to boolean
+        (datetime(1970, 1, 1), pl.Datetime("ms"), pl.Boolean, True, False),
+        (datetime(2022, 1, 1), pl.Datetime("ms"), pl.Boolean, True, True),
+        (datetime(1970, 1, 1), pl.Datetime("us"), pl.Boolean, True, False),
+        (datetime(2022, 1, 1), pl.Datetime("us"), pl.Boolean, True, True),
+        (datetime(1970, 1, 1), pl.Datetime("ns"), pl.Boolean, True, False),
+        (datetime(2022, 1, 1), pl.Datetime("ns"), pl.Boolean, True, True),
+        # date to boolean
+        (date(1970, 1, 1), pl.Date, pl.Boolean, True, False),
+        (date(2022, 1, 1), pl.Date, pl.Boolean, True, True),
+        # time to boolean
+        (time(hour=0), pl.Time, pl.Boolean, True, False),
+        (time(hour=1), pl.Time, pl.Boolean, True, True),
```

Comment:
These should all raise an `InvalidOperationError`. See https://github.com/pola-rs/polars/pull/14142

Reply:
None of them do on the latest main:

```python
>>> pl.Series([date(2024, 1, 1)]).cast(pl.Boolean)      # returns True
>>> pl.Series([datetime(2024, 1, 1)]).cast(pl.Boolean)  # returns True
>>> pl.Series([timedelta(hours=1)]).cast(pl.Boolean)    # returns True
>>> pl.Series([time(hour=1)]).cast(pl.Boolean)          # returns True
```

Is this an issue that should be raised?

Reply:
Also, I would argue that a `duration`, being a distance metric, should evaluate to False when zero, but I agree for the others that it should not.

Reply:
It will be fixed with the PR I linked.

Reply:
Ahh! It's an open PR, got it. I will remove the tests.